command/agent/retry_join.go

```diff
@@ -30,6 +35,57 @@ type DiscoverInterface interface {
 	Names() []string
 }
 
+// discoverProxy proxies retryJoiner discovery requests to either go-netaddrs or go-discover
+//
+// Addresses are first discovered using go-netaddrs, but if no addresses are able to be disocovered,
+// fall back to go-autodiscover
+type discoverProxy struct {
+	goDiscover *discover.Discover
+}
+
+// Addrs looks up and returns IP addresses using the method described by cfg.
+//
+// If cfg is a DNS name, IP addresses are looked up by querying the default
+// DNS resolver for A and AAAA records associated with the DNS name.
```

Comment:
This is a little misleading, so I want to call it out so it doesn't get copied into the eventual docs. The `go-netaddrs` library looks for the `exec=` prefix first and then if its not present, tries to resolve the name as if were a DNS entry.